### DISCUSSION TITLE
BackgroundTask vs task queue (Celery, RQ, etc)

### DISCUSSION THREAD

**Discussion by oTree-org** [encode/starlette#2398] [Q&A] [answered]
Hi, what is the general guidance on what type of tasks are appropriate for `BackgroundTask`? Is it just for quick things like sending an email? Or could it be used for tasks that take 1 minute - 1 hour? (either IO-bound like downloading files, or CPU/GPU compute-bound).

In my case my tasks are synchronous (not async).

**Comment by adriangb** [comment, marked as answer]
I would suggest considerably less than 30s which is the default SIGTERM timeout on Kuberneted (just general number). 1s to send an email should be fine, but I wouldn’t do 30s of compute intensive stuff.

**Reply by oTree-org** [reply]
Thanks!
